### PR TITLE
fp-hometask

### PR DESCRIPTION
**Pull request by Artemijacd**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -14,29 +14,74 @@ const mockedData: Row[] = rows.data;
...
+        })
+      
+        const fs2 = accounts.map((account) => {
+          return {
+            lastPayments: account.payments.length, 
+            posts: account.posts,
+          }
+        })
+
+        fs.map((item, i) => {
```
Comment (metam0rphus) [inline src/App.tsx:53]: accounts can have userID = 5 under index 1, and users can have userID = 7 under index 1

File: src/App.tsx [hunk 2]
```diff
@@ -14,29 +14,74 @@ const mockedData: Row[] = rows.data;
...
+            return rows.push({...item, ...fs1[i], ...fs2[i] });
+        })
+        return rows;
```
Comment (metam0rphus) [inline src/App.tsx:56]: The result doesn't correspond to Row[]

File: src/App.tsx [hunk 3]
```diff
@@ -14,29 +14,74 @@ const mockedData: Row[] = rows.data;
...
+            return {
+              username: user.username,
+              country: user.country,
+              name: user.name,
+            }
+        })
+      
+        const fs2 = accounts.map((account) => {
+          return {
+            lastPayments: account.payments.length,
```
Comment (metam0rphus) [inline src/App.tsx:48]: lastPayments is total sum of payments

File: src/App.tsx [hunk 4]
```diff
@@ -14,29 +14,74 @@ const mockedData: Row[] = rows.data;
...
+            return rows.push({...item, ...fs1[i], ...fs2[i] });
+        })
+        return rows;
+      };
+      setData(dataConverter(users, accounts, images));
+    });
   }, []);
 
+  const filterData = (data: Row[]) => {
+    if (data) {
```
Comment (metam0rphus) [inline src/App.tsx:63]: use "guard expression" to avoid excess nesting

File: src/App.tsx (hunk 1 above)
Comment (Artemijacd) [reply src/App.tsx]: fixed

**Pull request merged**